core/command/index.js

```diff
@@ -14,7 +15,7 @@ const cleanStack = require("./lib/clean-stack");
 const logPackageError = require("./lib/log-package-error");
 const warnIfHanging = require("./lib/warn-if-hanging");
 
-const DEFAULT_CONCURRENCY = 4;
+const DEFAULT_CONCURRENCY = os.cpus().length;
```

Comment:
To make the [tests](https://github.com/lerna/lerna/blob/5c1b84c699d80dbd8c5cce009aae8b82569e32a8/core/command/__tests__/command.test.js#L61-L89) pass, you'll need to mock out this result, otherwise it will always fail in environments other than your localhost (I have 12 logical cores on my laptop, for example).

```patch
diff --git a/core/command/__tests__/command.test.js b/core/command/__tests__/command.test.js
index 9cbbc60..0724038 100644
--- a/core/command/__tests__/command.test.js
+++ b/core/command/__tests__/command.test.js
@@ -9,6 +9,10 @@ const tempy = require("tempy");
 
 // partially mocked
 const ChildProcessUtilities = require("@lerna/child-process");
+const os = require("os");
+
+// normalize concurrency across different environments (localhost, CI, etc)
+jest.spyOn(os, "cpus").mockImplementation(() => new Array(42));
 
 // helpers
 const initFixture = require("@lerna-test/init-fixture")(__dirname);
@@ -70,14 +74,14 @@ describe("core-command", () => {
       const command = testFactory({ concurrency: "bla" });
       await command;
 
-      expect(command.concurrency).toBe(4);
+      expect(command.concurrency).toBe(42);
     });
 
     it("should fall back to default if concurrency given is 0", async () => {
       const command = testFactory({ concurrency: 0 });
       await command;
 
-      expect(command.concurrency).toBe(4);
+      expect(command.concurrency).toBe(42);
     });
 
     it("should fall back to 1 if concurrency given is smaller than 1", async () => {
```

Reply:
Thanks for the patch! I will apply it and test.